sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample09_MockingEventHubPublisher.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Producer;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    [TestFixture]
+    public class Sample09_MockingEventHubPublisher
+    {
+        /// <summary>
+        ///     Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        [Test]
+        public async Task EventBatch()
+        {
+            #region Snippet:EventHubs_Sample09_EventBatchMock
+
+            var size = 1024;
+            var store = new List<EventData> { new EventData(Array.Empty<byte>()), new EventData(Array.Empty<byte>()) };
+            var mockResponse = new Mock<Response>();
+            Response<EventDataBatch> response = Response.FromValue(EventHubsModelFactory.EventDataBatch(size, store), mockResponse.Object);
```

Comment:
The `Response` type is meant to represent  an HTTP response.   The Event Hubs clients don't use this, as they're AMQP-based.    I'm surprised this compiles, given that the return type of `CreateBatchAsync` is `ValueTask<EventDataBatch>`

What we'll want here is something more like:

```csharp
var mockEventBatch = EventHubsModelFactory.EventDataBatch(mockBatchSizeInBytes, eventStore);

mockEventHubProducerClient
    .Setup(c => c.CreateBatchAsync(It.IsAny<CancellationToken>()))
    .Returns(new ValueTask(mockEventBatch));
```